### PR TITLE
Depreciate proxy config binary

### PR DESCRIPTION
**Pull request by AllPurposeBen**
Some have noticed on 10.13+ machines, users see the system warning about 32-bit code `not being optimized of macOS...` in reference to the binary `proxy-config`, which is included in current builds of the Bluesky agent. 

This binary tool simply returns the url and port number of a configured _and_ enabled https proxy in system preferences. The providence of this binary is unknown but it's not unique in it's function. This PR introduces a code change to deprecate the external binary in favor of a check internal to the Bluesky code. 

it returns an enabled https proxy url in the same format as the binary did, eg. `https://10.0.1.1:8080/`

Validation that has been done:

- I have tested this code on OS's as a far back as 10.6.8 
- Agents **not** behind a proxy work as normal

Validation not done:

- Real world testing behind a functional https proxy server. 

Interesting note: In the current code, we're calling `proxy-config -s` to return a configured _https_ proxy but the url string it outputs always has `http://x.x.x.x:y` rather than `https://x.x.x.x:y`. That seems wrong to me but I guess it worked before. In the new code, I'm explicitly setting `https://`.

### CODE REVIEW

File: Client/bluesky.sh [hunk 1]
```diff
@@ -323,11 +323,23 @@ while [ "$activeNets" == "" ]; do
...
+# Get details on proxy settings. This can take some time so lets only ask the system once.
+proxyDetails=$(networksetup -getsecurewebproxy "$routeServiceName")
+# Are the proxy settings enabled?
+if [[ $(echo "$proxyDetails" | grep ^'Enabled: Yes') ]]; then
+	# It is enabled, lets read the data
+	proxyHostName=$(echo "$proxyDetails" | grep ^'Server:' | awk -F ': ' '{print $NF}')
+	proxyPort=$(echo "$proxyDetails" | grep ^'Port:' | awk -F ': ' '{print $NF}')
+	# sanity check, did we get those? A blank field for port in the GUI gives us an output string of "0", which is equally bad as null
+	if [[ -n $proxyHostName ]] && [[ -n $proxyPort ]] && [[ $proxyPort -ne 0 ]]; then
+		curlProxy="https://$proxyHostName:$proxyPort/"
```
Comment (bestmacs) [inline Client/bluesky.sh]: This needs to be http: not https: - the proxy server handles setting up a secure communication with the receiving server.  That seems weird to me too, but figure the client and the proxy are generally on the same LAN/behind the same firewall.  Plus as you noted, that was proxy-config's behavior too.

**Comment by sphen13** [comment]
i have not been able to test this in a true proxy environment - but if you think we are good @bestmacs and @AllPurposeBen, im willing to let it ride

File: Client/bluesky.sh (hunk 1 above)
Comment (AllPurposeBen) [reply Client/bluesky.sh]: Makes sense, switched to http in https://github.com/logicnow/BlueSky/pull/49/commits/7f0d2294f1026b8f9c5933ab357daaf382fb2e07

**Comment by sphen13** [comment]
FYI proxy-config is present and referenced within the temporary client as well. 

https://github.com/logicnow/BlueSky/blob/master/Admin%20Tools/temporary%20client/proxy-config

https://github.com/logicnow/BlueSky/blob/cfcb1667b24dabfe64def71dca1ac14163ad96f7/Admin%20Tools/temporary%20client/temporary%20client.applescript#L21

**Comment by AllPurposeBen** [comment]
> FYI proxy-config is present and referenced within the temporary client as well.

Hmmm...maybe rather than retooling the main script, I could just pull out that code into a standing replacement for old proxy-config. Putting that block of shell into the temp app Apple script could get...messy.

File: Client/bluesky.sh [hunk 2]
```diff
@@ -323,11 +323,23 @@ while [ "$activeNets" == "" ]; do
...
+# Get details on proxy settings. This can take some time so lets only ask the system once.
+proxyDetails=$(networksetup -getsecurewebproxy "$routeServiceName")
+# Are the proxy settings enabled?
+if [[ $(echo "$proxyDetails" | grep ^'Enabled: Yes') ]]; then
+	# It is enabled, lets read the data
+	proxyHostName=$(echo "$proxyDetails" | grep ^'Server:' | awk -F ': ' '{print $NF}')
+	proxyPort=$(echo "$proxyDetails" | grep ^'Port:' | awk -F ': ' '{print $NF}')
+	# sanity check, did we get those? A blank field for port in the GUI gives us an output string of "0", which is equally bad as null
+	if [[ -n $proxyHostName ]] && [[ -n $proxyPort ]] && [[ $proxyPort -ne 0 ]]; then
+		curlProxy="http://$proxyHostName:$proxyPort/"
```
Comment (sphen13) [inline Client/bluesky.sh]: actually looking at what we use curlProxy for i think this needs to be changed and treated as a curl option.

```suggestion
		curlProxy="-x http://$proxyHostName:$proxyPort"
```

Comment (AllPurposeBen) [reply Client/bluesky.sh]: Ok, I'll give you that one but before committing it back, my earlier question stands. If the answer is yes, this is moot.

**Comment by sphen13** [comment]
I hear yah. Thinking about it maybe it is best to just replace the existing proxy-config binary with a bash script that returns the proxy url if it’s enabled on the system. That should work without any real code change otherwise right?

Return nothing of proxy is not enabled - otherwise return http://proxyserver:proxyport

**Comment by AllPurposeBen** [comment]
Ok, this is supplanted by https://github.com/logicnow/BlueSky/pull/51, replacing the `proxy-config` binary whole without other changes to code that calls it.

**Pull request closed without merging**